File: people-whitelist/src/audit.js
export async function logAudit(pool, { actorEmail, action, subjectEmail, oldValue, newValue }) {
  await pool.query(
    `insert into audit_log (actor_email, action, subject_email, old_value, new_value)
     values ($1, $2, $3, $4, $5)`,
    [actorEmail || null, action, subjectEmail || null, oldValue || null, newValue || null]
  );
}

File: admin_api/audit.py
import json
from db import Db


def log_audit(actor_email: str, action: str, subject_email: str = None,
              old_value: dict = None, new_value: dict = None):
    with Db() as conn:
        with conn.cursor() as cur:
            cur.execute("""
                INSERT INTO audit_log (actor_email, action, subject_email, old_value, new_value)
                VALUES (%s, %s, %s, %s, %s)
            """, (
                actor_email, action, subject_email,
                json.dumps(old_value) if old_value else None,
                json.dumps(new_value) if new_value else None,
            ))
